images/virtualization-artifact/pkg/controller/vd/vd_reconciler.go

```diff
@@ -232,11 +232,21 @@ func (r *Reconciler) SetupController(_ context.Context, mgr manager.Manager, ctr
 		return fmt.Errorf("error setting watch on CVIs: %w", err)
 	}
 
-	w := watcher.NewVirtualDiskSnapshotWatcher(mgr.GetClient())
-	if err := w.Watch(mgr, ctr); err != nil {
+	snapshotWatcher := watcher.NewVirtualDiskSnapshotWatcher(mgr.GetClient())
+	if err := snapshotWatcher.Watch(mgr, ctr); err != nil {
 		return fmt.Errorf("error setting watch on VDSnapshots: %w", err)
 	}
 
+	viFromVDWatcher := watcher.NewVirtualImageWatcher()
```

Comment:
There is already watchers for vi and cvi (watchers.NewObjectRefWatcher)

Reply:
No, this is other watcher.

Reply:
Fixed.